extensions/standard-processors/processors/PutTCP.cpp

```diff
@@ -0,0 +1,582 @@
+    ->asType<core::DataSizeValue>()
+    ->build();
+
+const core::Relationship PutTCP::Success{"success", "FlowFiles that are sent to the destination are sent out this relationship."};
+const core::Relationship PutTCP::Failure{"failure", "FlowFiles that encountered IO errors are send out this relationship."};
+
+constexpr size_t chunk_size = 1024;
+
+PutTCP::PutTCP(const std::string& name, const utils::Identifier& uuid)
+    : Processor(name, uuid) {}
+
+PutTCP::~PutTCP() = default;
+
+void PutTCP::initialize() {
+  setSupportedProperties(properties());
+  setSupportedRelationships(relationships());
+}
+
+void PutTCP::notifyStop() {}
+
+void PutTCP::onSchedule(core::ProcessContext* const context, core::ProcessSessionFactory*) {
+  gsl_Expects(context);
+
+  // if the required properties are missing or empty even before evaluating the EL expression, then we can throw in onSchedule, before we waste any flow files
+  if (context->getProperty(Hostname).value_or(std::string{}).empty()) {
+    throw Exception{ExceptionType::PROCESSOR_EXCEPTION, "missing hostname"};
+  }
+  if (context->getProperty(Port).value_or(std::string{}).empty()) {
+    throw Exception{ExceptionType::PROCESSOR_EXCEPTION, "missing port"};
+  }
+  if (auto idle_connection_expiration = context->getProperty<core::TimePeriodValue>(IdleConnectionExpiration); idle_connection_expiration && idle_connection_expiration->getMilliseconds() > 0ms)
+    idle_connection_expiration_ = idle_connection_expiration->getMilliseconds();
+  else
+    idle_connection_expiration_.reset();
+
+  if (auto timeout = context->getProperty<core::TimePeriodValue>(Timeout); timeout && timeout->getMilliseconds() > 0ms)
+    timeout_ = timeout->getMilliseconds();
+  else
+    timeout_ = 15s;
+
+  std::string context_name;
+  ssl_context_service_.reset();
+  if (context->getProperty(SSLContextService.getName(), context_name) && !IsNullOrEmpty(context_name)) {
+    if (auto controller_service = context->getControllerService(context_name)) {
+      ssl_context_service_ = std::dynamic_pointer_cast<minifi::controllers::SSLContextService>(context->getControllerService(context_name));
+      logger_->log_error("%s is not a SSL Context Service", context_name);
+    } else {
+      logger_->log_error("Invalid controller service: %s", context_name);
+    }
+  }
+
+  delimiter_ = utils::span_to<std::vector>(gsl::make_span(context->getProperty(OutgoingMessageDelimiter).value_or(std::string{})).as_span<const std::byte>());
+
+  if (context->getProperty<bool>(ConnectionPerFlowFile).value_or(false))
+    connections_.reset();
+  else
+    connections_.emplace();
+
+  if (auto max_size_of_socket_send_buffer = context->getProperty<core::DataSizeValue>(MaxSizeOfSocketSendBuffer))
+    max_size_of_socket_send_buffer_ = max_size_of_socket_send_buffer->getValue();
+  else
+    max_size_of_socket_send_buffer_.reset();
+}
+
+namespace {
+template<class SocketType>
+class ConnectionHandler : public IConnectionHandler {
```

Comment:
I agree with you, but that naming would suggest that it can be used as non-NetworkConnectionHandler or non-TCPConnectiotionHandler, I've changed the IConnectionHandler to ConnectionHandlerBase (that should be more in line with our naming conventions), what do you think? https://github.com/apache/nifi-minifi-cpp/pull/1419/commits/8b7f14e569f6b60aada1fa9488102f1b9bbb92ec

Reply:
I think the name of the ConnectionHandler should only suggest the interface it represents, so I don't think it's a problem if it could be used for other connections as well that are not network specific connections like an IPC connection. That could also implement the same interface (even though we will probably never have that). It's better to see in the naming what is more specific in the inherited classes even if it just implements a pure virtual interface.

But in this specific case I don't strongly insist on this, I'm okay with ConnectionHandlerBase here, it's still better than IConnectionHandler.

Reply:
I think as long as it remains in the PutTCP implementation file, the name doesn't have to include info that can be deduced from the context. What do you think?

Reply:
I think you are right, in this context it doesn't really matter, this is more of a philosophical clean coding question :)